src/safeds/data/tabular/transformation/_table_transformer.py

```diff
@@ -21,7 +23,16 @@ def __hash__(self) -> int:
         hash : int
             The hash value.
         """
-        return xxhash.xxh3_64(self.__class__.__qualname__.encode("utf-8") + (1 if self.is_fitted() else 0).to_bytes(1)).intdigest()
+        added = self.get_names_of_added_columns() if self.is_fitted() else []
+        changed = self.get_names_of_changed_columns() if self.is_fitted() else []
+        removed = self.get_names_of_removed_columns() if self.is_fitted() else []
+        return xxhash.xxh3_64(
+            self.__class__.__qualname__.encode("utf-8")
+            + (1 if self.is_fitted() else 0).to_bytes(1)
+            + functools.reduce(operator.add, [feature.encode("utf-8") for feature in added], b"A")
+            + functools.reduce(operator.add, [feature.encode("utf-8") for feature in changed], b"C")
+            + functools.reduce(operator.add, [feature.encode("utf-8") for feature in removed], b"R"),
+        ).intdigest()
```

Comment:
It seems like your pattern for creating a hash is always the same:

1. `None` -> `b"\0"`
2. `int` -> `to_bytes`
3. `bool` -> `to 0/1`, then `to_bytes`
4. `string` -> `encode`
5. `float` -> `struct` pack
6. `list` (or some other iterable) -> `reduce` + `add` $^1$
7. other -> `hash`

Thus, I'd strongly suggest capturing this logic in an extracted method, so it's easier to add hashing for other conceptually immutable classes:

* Create a `_utils` package in `src/safeds` next to `_config`.
* Add a module `_hashing.py` in there.
* Add a function `hash(*value: any) -> int` in there

$^1$ The different initial values for `reduce` (here `b"A"`) could either be omitted or we keep track of a counter when we hash the arguments passed to `hash`.

Reply:
Will do :+1:

Reply:
Maybe `structuralHash` would be fitting names (like [structural equality](https://kotlinlang.org/docs/equality.html#structural-equality) vs [referential equality](https://kotlinlang.org/docs/equality.html#referential-equality)).